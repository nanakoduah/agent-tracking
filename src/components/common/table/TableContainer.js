import React, { Component } from 'react';
import PropTypes from 'prop-types';

class TableContainer extends Component {
  static propTypes = {
    component: PropTypes.func.isRequired
  };
  state = {};

  render() {
    const { component: Component } = this.props;
    return <Component />;
  }
}

export default TableContainer;
