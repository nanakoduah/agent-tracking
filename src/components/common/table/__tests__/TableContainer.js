import React from 'react';
import renderer from 'react-test-renderer';

import TableContainer from '../TableContainer';

describe('TableContainer', () => {
  it('should initialize properly', () => {
    const tree = renderer.create(
      <TableContainer component={jest.fn(() => null)} />
    );
    expect(tree.getInstance().state).toMatchSnapshot();
  });
});
