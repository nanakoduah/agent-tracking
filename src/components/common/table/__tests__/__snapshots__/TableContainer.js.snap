// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableContainer should initialize properly 1`] = `Object {}`;
